ambari-server/src/main/java/org/apache/ambari/server/upgrade/UpgradeCatalog300.java

```diff
@@ -784,9 +888,50 @@ protected void updateKerberosConfigurations() throws AmbariException {
                   true, false);
             }
           }
+          if (doPrincipalsCreation) {
```

Comment:
rather that use a Boolean argument for this, wouldn't it be better it use a mocked `PrepareKerberosIdentitiesServerAction` instance in the injector?  

Then you can do
```
PrepareKerberosIdentitiesServerAction prepareIdentities = injector.getInstance(PrepareKerberosIdentitiesServerAction.class)
```

Rather than
```
PrepareKerberosIdentitiesServerAction prepareIdentities = new PrepareKerberosIdentitiesServerAction();
...
// inject whatever we need for calling desired server action
injector.injectMembers(prepareIdentities);
```

And you do not have to introduce an argument used only for testing purposes - which is never actually set to false.